src/Telemetry/UserLevelCacheWriter.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Telemetry
+{
+    public sealed class UserLevelCacheWriter
+    {
+        private const string AzureFunctionsSqlBindingsProfileDirectoryName = ".azurefunctions-sqlbindings";
+        private readonly string _azureFunctionsSqlBindingsTryUserProfileFolderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), AzureFunctionsSqlBindingsProfileDirectoryName);
+
+        public string RunWithCache(string cacheKey, Func<string> getValueToCache)
+        {
+            string cacheFilepath = this.GetCacheFilePath(cacheKey);
+            try
+            {
+                if (!File.Exists(cacheFilepath))
+                {
+                    if (!Directory.Exists(this._azureFunctionsSqlBindingsTryUserProfileFolderPath))
+                    {
+                        Directory.CreateDirectory(this._azureFunctionsSqlBindingsTryUserProfileFolderPath);
+                    }
+
+                    string runResult = getValueToCache();
+
+                    File.WriteAllText(cacheFilepath, runResult);
+                    return runResult;
+                }
+                else
+                {
+                    return File.ReadAllText(cacheFilepath);
+                }
+            }
+            catch (Exception ex)
+            {
+                if (ex is UnauthorizedAccessException
```

Comment:
Curious about your thoughts around adding a telemetry event here to track issues in this codepath

Reply:
It's kind of a catch-22 situation there because we need these common properties to do logging currently. I could refactor it to send a special event in this case, but I don't see that being something worth the effort at this time unless we think this is something that is more likely to happen.

Reply:
Gotcha. Yeah, I'm most worried about deployed scenarios. I just don't know that much about how permissions are setup in Azure, and whether even the user dir is locked down.

Reply:
It's something I need to look at anyways - it feels like we need to have a flag for when it's running in Azure. I'm going to investigate what that might look like (hopefully some env var I can use?)

Reply:
Looked into it and it was pretty easy to pipe through so I went ahead and added some exception logging here.